### PR TITLE
External sources on no results page

### LINKED ISSUE #1999: The external sources on the No results form should not be in a popover
## Problem

The external search feature shown on the "No results" page should show a row (or column for mobile) of button links to the external sources.

Screenshot 2022-11-21 at 4 56 38 PM
Screenshot 2022-11-21 at 5 19 18 PM

Figma link desktop: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Audio-Release?node-id=1479%3A20812&t=JjmB63Y8kMIIlwm3-4
Figma link mobile: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Audio-Release?node-id=1479%3A20846&t=3szwykXToMs7gluh-4

## Description

We need to refactor`VExternalSourcesList` so that the buttons can be used both in a popover and in a row/column. Then, we should replace the popover component on the No results list with the button row/column for the external sources.

## Additional context

_Split from https://github.com/WordPress/openverse-frontend/issues/734_

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1999 by @obulat

## Description

This PR refactors the No results page to use rows/column of buttons.

@panchovm, there are no updated mockups for this in Figma, so I'll need your input on paddings and sizes.

## Testing Instructions

Run the app, and try searching for something that doesn't have any results, something like "lskdjflskdjflsk". This should open a "No results" page. Here, on desktop, the external sources should be in a row of buttons, and on mobile, in a column.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — COMMENTED** [review]
Could you add some visual regression tests of the component stories (if any) and the page.

**Review by panchovm — CHANGES_REQUESTED** [review]
The change looks great, and since there is no mockup, I suggest applying two changes:

* **Text style:** Apply the "Label bold" style as defined in the Design Library.
* **Spacing:** Set a `8px` gap between label and icon.

Sorry for replying late as I was reviewing the buttons created in the Design Library and in storybook to select the correct one. However, there are multiple inconsistencies between both sources that do not block this PR. Applying the changes described above is sufficient.

In parallel, I started exploring a layout improvement for this and Timeout pages that will share in a separate design ticket.

File: src/components/VErrorSection/VNoResults.vue [hunk 1]
```diff
@@ -1,35 +1,56 @@
...
-    <VExternalSourceList class="mt-4 md:mt-6" :type="type" :query="query" />
+
+    <div class="mt-4 flex flex-col flex-wrap gap-4 gap-2 md:mt-6 md:flex-row">
+      <VButton
+        v-for="source in externalSources"
+        :key="source.name"
+        as="VLink"
+        :href="source.url"
+        variant="secondary-bordered"
+        class="text-dark-charcoal"
```
Comment (obulat) [inline src/components/VErrorSection/VNoResults.vue:17]:
```suggestion
        class="text-dark-charcoal label-bold gap-x-2"
```

**Review by dhruvkb — APPROVED** [review]
The changes look good to me!

I have some issues with the look of full-width buttons at the tablet breakpoint but that's a design issue and not something to block the PR over. @panchovm maybe more than one column or just the individual buttons as seen on the desktop would be better.

**Comment by panchovm** [comment]
Thanks for the suggestion @dhruvkb I am currently working on a proposal for updating this and Timeout pages' layouts. For this PR scope, the buttons look correct.

I plan to share the layout proposal soon.

**Review by zackkrida — APPROVED** [review]
LGTM. The spacing between the text and the buttons on mobile feels too tight to me, but we can refine the design in a subsequent PR.

CleanShot 2023-01-12 at 08 53 02

File: src/components/VErrorSection/VNoResults.vue [hunk 2]
```diff
@@ -1,35 +1,56 @@
...
   <div class="no-results text-center md:text-left">
-    <h1 class="break-words text-4xl md:text-6xl">
-      {{ $t("no-results.heading", { query: query.q }) }}
+    <h1 class="heading-4 md:heading-2 break-words">
+      {{ $t("no-results.heading", { query: searchTerm }) }}
     </h1>
-    <h2
-      class="mt-10 text-base font-normal md:mt-16 md:text-3xl md:font-semibold"
-    >
+    <h2 class="description-regular md:heading-5 mt-10 md:mt-16">
```
Comment (panchovm) [inline src/components/VErrorSection/VNoResults.vue:6]: I would replace the spacing `10` and `16` with `4` (16px) in all breakpoints to reduce the gap between both headings

File: src/components/VErrorSection/VNoResults.vue [hunk 3]
```diff
@@ -1,35 +1,56 @@
...
       {{ $t("no-results.alternatives") }}
     </h2>
-    <VExternalSourceList class="mt-4 md:mt-6" :type="type" :query="query" />
+
+    <div class="mt-4 flex flex-col flex-wrap gap-4 gap-2 md:mt-6 md:flex-row">
```
Comment (panchovm) [inline src/components/VErrorSection/VNoResults.vue:10]: Similar to my previous comment, I suggest replacing spacing `4` and `6` with `10` (40px) to give more space between headings and buttons list in all breakpoints.

**Review by panchovm — CHANGES_REQUESTED** [review]
I suggested two spacing tweaks in the layout that have nothing to do with the button's style.

For @zackkrida's point. Yes, the button style is something we need to tackle in the future to address the spacing consistency across the site.

**Pull request merged**